CMakeLists.txt

```diff
@@ -82,21 +82,27 @@ if (WIN32)
     set_property(DIRECTORY APPEND PROPERTY COMPILE_DEFINITIONS $<$<CONFIG:RELWITHDEBINFO>:${Definition}>)
   endforeach (Definition ${CLR_DEFINES_RELWITHDEBINFO_INIT})
 
-elseif (CLR_CMAKE_PLATFORM_UNIX)  
+elseif (CLR_CMAKE_PLATFORM_UNIX)
+  # Set the values to display when interactively configuring CMAKE_BUILD_TYPE
+  set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "DEBUG;RELEASE;RELWITHDEBINFO")
+
+  # Use uppercase CMAKE_BUILD_TYPE for the string comparisons below
+  string(TOUPPER ${CMAKE_BUILD_TYPE} UPPERCASE_CMAKE_BUILD_TYPE)
+
   # For single-configuration toolset
   # set the different configuration defines.
-  if (CMAKE_BUILD_TYPE STREQUAL DEBUG)
+  if (UPPERCASE_CMAKE_BUILD_TYPE STREQUAL DEBUG)
     # First DEBUG
     set_property(DIRECTORY  PROPERTY COMPILE_DEFINITIONS ${CLR_DEFINES_DEBUG_INIT})
-  elseif (CMAKE_BUILD_TYPE STREQUAL RELEASE)
+  elseif (UPPERCASE_CMAKE_BUILD_TYPE STREQUAL RELEASE)
     # Then RELEASE
     set_property(DIRECTORY PROPERTY COMPILE_DEFINITIONS ${CLR_DEFINES_RELEASE_INIT})
-  elseif (CMAKE_BUILD_TYPE STREQUAL RELWITHDEBINFO)
+  elseif (UPPERCASE_CMAKE_BUILD_TYPE STREQUAL RELWITHDEBINFO)
     # And then RELWITHDEBINFO
     set_property(DIRECTORY APPEND PROPERTY COMPILE_DEFINITIONS ${CLR_DEFINES_RELWITHDEBINFO_INIT})
-  else (CMAKE_BUILD_TYPE STREQUAL RELWITHDEBINFO)
-    message(FATAL_ERROR "Unknown build type! Allowed DEBUG, RELEASE and RELWITHDEBINFO!")
-  endif (CMAKE_BUILD_TYPE STREQUAL DEBUG)
+  else (UPPERCASE_CMAKE_BUILD_TYPE STREQUAL RELWITHDEBINFO)
```

Comment:
I would suggest removing the "(UPPERCASE_CMAKE_BUILD_TYPE STREQUAL RELWITHDEBINFO)" from the "else" so that the change is inline with what @mathstuf is working upon.

Otherwise, LGTM.